File: scratch/fix_human.py
import re

path = 'frontend/src/components/triage/HumanBodyModel.tsx'
with open(path, 'r', encoding='utf-8') as f:
    content = f.read()

content = re.sub(r'\s*\{\/\* @ts-expect-error - Auto-suppressed during migration \*\/\}', '', content)
content = re.sub(r'\s*// @ts-expect-error - Auto-suppressed during migration', '', content)

content = re.sub(
    r'export interface HumanBodyModelProps \{\n  onPartClick\?\: \(\.\.\.args\: DynamicStateObject\[\]\) => void;\n    \[key: string\]: ReturnType<typeof JSON\.parse>;\n\}',
    r'''export interface HumanBodyModelProps {
  onPartClick?: (part: string) => void;
}''',
    content
)

content = content.replace('e: DynamicStateObject', 'e: any')
content = content.replace('import { DynamicStateObject } from "./../../types/DynamicState";\n', '')
content = content.replace('onPartClick(', 'onPartClick?.(')

with open(path, 'w', encoding='utf-8') as f:
    f.write(content)

File: scratch/fix_human.js
const fs = require('fs');

const path = 'frontend/src/components/triage/HumanBodyModel.tsx';
let content = fs.readFileSync(path, 'utf8');

content = content.replace(/\s*\{\/\*\s*@ts-expect-error\s*-\s*Auto-suppressed during migration\s*\*\/\}/g, '');
content = content.replace(/\s*\/\/\s*@ts-expect-error\s*-\s*Auto-suppressed during migration/g, '');

content = content.replace(
  'export interface HumanBodyModelProps {\n  onPartClick?: (...args: DynamicStateObject[]) => void;\n    [key: string]: ReturnType<typeof JSON.parse>;\n}',
  'export interface HumanBodyModelProps {\n  onPartClick?: (part: string) => void;\n}'
);
content = content.replace(/e: DynamicStateObject/g, 'e: any');
content = content.replace('onPartClick("', 'onPartClick?.("');

content = content.replace(/import \{ DynamicStateObject \} from "\.\.\/\.\.\/types\/DynamicState";\n/g, '');

// Prepend the triple slash reference
content = '/// <reference types="@react-three/fiber" />\n' + content;

fs.writeFileSync(path, content, 'utf8');
